src/credentials/providers/sharedCredentialsProvider.ts

```diff
@@ -87,7 +104,10 @@ export class SharedCredentialsProvider implements CredentialsProvider {
     public validate(): string | undefined {
         const expectedProperties: string[] = []
 
-        if (hasProfileProperty(this.profile, SHARED_CREDENTIAL_PROPERTIES.ROLE_ARN)) {
+        if (hasProfileProperty(this.profile, SHARED_CREDENTIAL_PROPERTIES.CREDENTIAL_SOURCE)) {
+            // Additional properties are not required
+            return undefined
```

Comment:
We should check for the existence of `source_profile` since the two properties cannot coexist:

> You cannot specify both credential_source and source_profile in the same profile. 


Reference: https://docs.aws.amazon.com/sdkref/latest/guide/setting-global-credential_source.html